src/library/scala/concurrent/BatchingExecutor.scala

```diff
@@ -26,7 +26,13 @@ trait Batchable {
 
 private[concurrent] object BatchingExecutorStatics {
   final val emptyBatchArray: Array[Runnable] = new Array[Runnable](0)
-  final val marker = ""
+
+  // Max number of Runnables executed nested before starting to batch (to prevent stack exhaustion)
+  final val syncPreBatchDepth = 16
+
+  // Max number of Runnables processed in one go (to prevent starvation of other tasks on the pool)
+  final val runLimit = 1024
```

Comment:
These limits are up for debate, I'm not sure that there is any optimal single number at all. Could lower to 64 or 128 or whatnot.